crates/bevy_sprite/src/mesh2d/mesh.rs

```diff
@@ -39,28 +39,70 @@ use bevy_render::{
     },
     view::{
         ExtractedView, ViewTarget, ViewUniform, ViewUniformOffset, ViewUniforms, ViewVisibility,
+        Visibility,
     },
     Extract, ExtractSchedule, Render, RenderApp, RenderSet,
 };
-use bevy_transform::components::GlobalTransform;
+use bevy_transform::components::{GlobalTransform, Transform};
 use bevy_utils::tracing::error;
 use nonmax::NonMaxU32;
 
 use crate::Material2dBindGroupId;
 
-/// Component for rendering with meshes in the 2d pipeline, usually with a [2d material](crate::Material2d) such as [`ColorMaterial`](crate::ColorMaterial).
+/// A component for rendering 2D meshes, typically with a [`MeshMaterial2d`] using a [`ColorMaterial`].
 ///
-/// It wraps a [`Handle<Mesh>`] to differentiate from the 3d pipelines which use the handles directly as components
-#[derive(Default, Clone, Component, Debug, Reflect, PartialEq, Eq, Deref, DerefMut)]
-#[reflect(Default, Component, Debug, PartialEq)]
-pub struct Mesh2dHandle(pub Handle<Mesh>);
-
-impl From<Handle<Mesh>> for Mesh2dHandle {
+/// Meshes without a [`MeshMaterial2d`] will be rendered with a [default material].
+///
+/// [`MeshMaterial2d`]: crate::MeshMaterial2d
+/// [`ColorMaterial`]: crate::ColorMaterial
+/// [default material]: crate::MeshMaterial2d#default-material
+///
+/// # Example
+///
+/// ```
+/// # use bevy_sprite::{ColorMaterial, Mesh2d, MeshMaterial2d};
+/// # use bevy_ecs::prelude::*;
+/// # use bevy_render::mesh::Mesh;
+/// # use bevy_color::palettes::basic::RED;
+/// # use bevy_asset::{AssetServer, Assets};
+/// # use bevy_math::primitives::Circle;
+/// #
+/// // Spawn an entity with a mesh using `ColorMaterial`.
+/// fn setup(
+///     mut commands: Commands,
+///     mut meshes: ResMut<Assets<Mesh>>,
+///     mut materials: ResMut<Assets<ColorMaterial>>,
+///     asset_server: Res<AssetServer>
+/// ) {
+///     commands.spawn((
+///         Mesh2d(meshes.add(Circle::new(50.0))),
+///         MeshMaterial2d(materials.add(ColorMaterial::from_color(RED))),
+///     ));
+/// }
+/// ```
+#[derive(Component, Clone, Debug, Default, Deref, DerefMut, Reflect, PartialEq, Eq)]
+#[reflect(Component, Default)]
+#[require(Transform, Visibility)]
+pub struct Mesh2d(pub Handle<Mesh>);
```

Comment:
I think we should probably move this into `bevy_render` for symmetry with Mesh3d. Then they would both follow the same pattern (core type in bevy_render, render implementation in respective sprite/pbr crate).

Reply:
Done!